### PR TITLE
DOTS compatible error/loading shaders GFXHYR-308

### PR DESCRIPTION
**Pull request by vincent-breysse**
### /!\ Most of the changes are test related

# Purpose of this PR

#### URP changes only
- Merge `FallbackError.shader` and `MaterialError.shader`. These shaders were originally split to have a DOTS compatible error shader while still having a shader being compatible with built-in RP.
However this PR tries to simplify this by having only one shader with two sub shaders and use the `Tags { "RenderPipeline" = "UniversalPipeline" }` thingy

#### URP/HDRP changes
- Add `#pragma editor_sync_compilation` to error/loading shaders to avoid the "error/loading shader is currently loading" edge case
- Add DOTS compatible loading shader
- Some `[Reload("Path/To/Shader.shader")]` additions/updates

These new shaders are needed by Hybrid Renderer since built-in error/loading shaders aren't DOTS compatible. 

# Testing status
#### - Added URP/HDRP scenes to test various error shader cases when rendering with a BRG (not automated) :
    - Compilation error in HLSL code
    - Compilation error in shader parser (different internal handling than an error in HLSL code)
    - Shader with missing DOTS_INSTANCING_ON keyword
    - Material with missing shader
    - MeshRenderer with missing material
    - MeshRenderer with null material
    - Shader non SRP Batcher compatible
    - Shader not supported
    - Shader pass not supported (different internal handling than whole shader not being supported)
#### - Yamato PR URP/HDRP on trunk green except:
    Failing on master as well:
    - URP_Lighting on Win_Vulkan_Standalone_mono_Linear on version trunk
    - HDRP on Win_DX11_playmode_mono_Linear on version trunk
    - HDRP on Win_DX11_playmode_XR_mono_Linear on version trunk
    - HDRP on Win_DX12_playmode_mono_Linear on version trunk
    - HDRP on Win_Vulkan_playmode_mono_Linear on version trunk
    - HDRP on OSX_Metal_playmode_mono_Linear on version trunk
    - HDRP on Linux_Vulkan_playmode_mono_Linear on version trunk
    - HDRP_DXR on Win_DX12_playmode_mono_Linear on version trunk

### CODE REVIEW

**Comment by vincent-breysse** [comment]
Waiting for reviews before asking for QA passes

**Comment by vincent-breysse** [comment]
@cinight @TomasKiniulis There are a few things you need to know to test the PR correctly. I'm going to give you more info on Slack. Please don't start testing before I ping you

**Review by JussiKnuuttila — APPROVED** [review]

File: TestProjects/BatchRendererGroup_URP/ProjectSettings/EditorSettings.asset [hunk 1]
```diff
@@ -22,7 +22,7 @@ EditorSettings:
     inProgressEnabled: 1
   m_EnableTextureStreamingInEditMode: 1
   m_EnableTextureStreamingInPlayMode: 1
-  m_AsyncShaderCompilation: 0
+  m_AsyncShaderCompilation: 1
```
Comment (sebastienlagarde) [inline TestProjects/BatchRendererGroup_URP/ProjectSettings/EditorSettings.asset:25]: Are you sure about this one? Async compilation is usually disable in test project as it was causing various issue with uncompiled shader and also with SRP batcher. Any reasons to enable it?

**Review by sebastienlagarde — APPROVED** [review]
small comment otherwise good for hdrp section

File: TestProjects/BatchRendererGroup_URP/ProjectSettings/EditorSettings.asset (hunk 1 above)
Comment (vincent-breysse) [reply TestProjects/BatchRendererGroup_URP/ProjectSettings/EditorSettings.asset:25]: Yes, it should be good. This PR is only the Graphics part of the error/loading material implementation. It is dependent on the associated C++ PR which aims to provide good support for BRG error/loading material and also improve SRP Batcher behaviour when dealing with pathological shaders (e.g shader not compiled, shader compilation errors, not SRP Batcher compatible, missing `DOTS_INSTANCING_ON`, etc). Most of the changes are located in the C++ PR.

**Review by TomasKiniulis — APPROVED** [review]
Tested BatchRendererGroup_HDRP, Amalienborg palace builds on Deferred and Forward with Game Objects and BRG. Looks good on HDRP side. 

Test doc for more details: https://confluence.unity3d.com/pages/viewpage.action?pageId=172268208

**Review by cinight — APPROVED** [review]
All good. Test Doc: https://docs.google.com/document/d/1QAfxaWR3YDvCPR6pbvTniqassj1ETHdD-j94cfoUCVs/edit#

**Review by phi-lira — COMMENTED** [review]
LGTM.

File: com.unity.render-pipelines.universal/Shaders/Utils/FallbackLoading.shader [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    {
+        Pass
+        {
+            CGPROGRAM
+            #pragma vertex vert
+            #pragma fragment frag
+            #pragma target 2.0
+            #pragma multi_compile _ UNITY_SINGLE_PASS_STEREO STEREO_INSTANCING_ON STEREO_MULTIVIEW_ON
+            #pragma editor_sync_compilation
+            #include "UnityCG.cginc"
```
Comment (phi-lira) [inline com.unity.render-pipelines.universal/Shaders/Utils/FallbackLoading.shader:59]: I don't understand why this sub shader is for Builtin not URP. We should have a fallback to display the loading error in URP for platforms that dont support SM 4.5.

File: com.unity.render-pipelines.universal/Shaders/Utils/FallbackError.shader [hunk 3]
```diff
@@ -1,6 +1,51 @@
...
+
+            float4 frag (v2f i) : SV_Target
+            {
+                return float4(1,0,1,1);
+            }
+            ENDHLSL
+        }
+    }
+
     SubShader
```
Comment (phi-lira) [inline com.unity.render-pipelines.universal/Shaders/Utils/FallbackError.shader:49]: Iirc correctly you need another subshader here for target 2.0 with Universal Render Pipeline tag. So there's an error when running in both SM 4.5, 2.0 for URP and one for builtin.

File: com.unity.render-pipelines.universal/Shaders/Utils/FallbackLoading.shader (hunk 2 above)
Comment (vincent-breysse) [reply com.unity.render-pipelines.universal/Shaders/Utils/FallbackLoading.shader]: This sub shader is mostly here for consistency and isn't actually strictly needed. I removed it in the latest push as agreed on Slack

File: com.unity.render-pipelines.universal/Shaders/Utils/FallbackError.shader (hunk 3 above)
Comment (vincent-breysse) [reply com.unity.render-pipelines.universal/Shaders/Utils/FallbackError.shader:50]: This built-in RP SM 2.0 sub shader is needed to handle the case where you're trying to render URP materials without having set the URP Asset in the project settings beforehand. As agreed on Slack, I just added a comment about this to explain why this sub shader does exist.

**Review by phi-lira — APPROVED** [review]

**Pull request merged**